corporate/lib/stripe.py

```diff
@@ -587,13 +587,22 @@ def invoice_plans_as_needed(event_time: datetime=timezone_now()) -> None:
     for plan in CustomerPlan.objects.filter(next_invoice_date__lte=event_time):
         invoice_plan(plan, event_time)
 
-def attach_discount_to_realm(realm: Realm, discount: Decimal) -> None:
+def attach_discount_to_realm(realm: Realm, discount: Decimal, acting_user: Optional[UserProfile]=None) -> None:
+    try:
+        old_discount: Optional[Decimal] = realm.customer.default_discount
+    except AttributeError:
+        old_discount = None
```

Comment:
Don't access realm.customer directly. It will throw `RelatedObjectDoesNotExist` error if there is no customer associated with the realm object. Instead use `get_customer_by_realm` to access the customer. And then use that to get `old_discount` if customer is not None.

Also update `Customer.objects.update_or_create(realm=realm, defaults={'default_discount': discount})` since we no longer need to try to create a customer object if the above query returned a customer. So update the value of default_discount if the customer is not None and create a new customer and set the default_discount using objects.create otherwise.